### PR TITLE
Add Block-editor-keyboard-shortcut test case

### PR DESCRIPTION
**Pull request by alvitazwar**
## What?
Based on https://github.com/WordPress/gutenberg/pull/38570, part of https://github.com/WordPress/gutenberg/issues/38851. Migrate [block-editor-keyboard-shortcuts.test.js](https://github.com/WordPress/gutenberg/blob/trunk/packages/e2e-tests/specs/editor/various/block-editor-keyboard-shortcuts.test.js)

## Why?
See https://github.com/WordPress/gutenberg/pull/38570 for its background and rationale.

This is split into a new PR for easier review.

## How?
See https://github.com/WordPress/gutenberg/pull/38570 for the proposed migration steps.

## Testing Instructions
npm run test:e2e:playwright /test/e2e/specs/editor/various/block-editor-keyboard-shortcuts.spec.js
## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/55917380/381ed206-721a-4be7-b8b1-9015b9bda964

### CODE REVIEW

**Comment by kevin940726** [comment]
Hey there! I wanted to express my gratitude for the PR you submitted. By the way, have you had a chance to check out the latest best practices guide? It's packed with some really helpful tips and tricks that can make a big difference in how we write reliable and readable e2e tests. One key suggestion is to make good use of the `getByRole` method when selecting elements, and consider using `editor.getBlocks` instead of snapshot testing. It can really streamline our workflow. Thanks again for your contribution!

**Comment by alvitazwar** [comment]
@kevin940726  Can you please recheck that for me?

**Comment by kevin940726** [comment]
Hi @alvitazwar! I've been busy with something else recently, but this is still on my backlog! I'll get back to it as soon as possible 🙇 .

File: test/e2e/specs/editor/various/block-editor-keyboard-shortcuts.spec.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	} ) => {
+		await page.getByRole( 'button', { name: 'Add default block' } ).click();
+		await page.keyboard.type( '1st' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '2nd' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '3rd' );
+		await expect.poll( editor.getEditedPostContent ).toBe( MatchObject );
+		await page.keyboard.press( 'ArrowUp' );
+		page.locator( '[aria-label="Multiple selected blocks"]' );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/block-editor-keyboard-shortcuts.spec.js:65]: Call the just the locator doesn't do anything. This should be converted into a proper assertion.

**Review by Mamaduka — CHANGES_REQUESTED** [review]
The PR removes the `specs/editor/various/autosave.test.js` file instead of the `block-editor-keyboard-shortcuts.test.js`.

**Additional notes:**
* I'm not sure if `await expect.poll( editor.getEditedPostContent ).toBe( MatchObject );` assertions check anything related to keyboard navigation.
* Let's combine two `test.describe` into one. The grouping usually makes it harder to know where the new tests go. I think it's better to split tests into multiple files if there's a need for a separate test group.

**Pull request closed without merging**